stores/yaml/store.go

```diff
@@ -118,45 +122,89 @@ func (store *Store) LoadEncryptedFile(in []byte) (sops.Tree, error) {
 		}
 	}
+
+	var branches sops.TreeBranches
+	for _, doc := range data {
+		for i, item := range doc {
+			if item.Key == "sops" { // Erase
+				doc = append(doc[:i], doc[i+1:]...)
+			}
+		}
+		branches = append(branches, store.mapSliceToTreeBranch(doc))
+	}
 	return sops.Tree{
-		Branch:   store.mapSliceToTreeBranch(data),
+		Branches: branches,
 		Metadata: metadata,
 	}, nil
 }
 
-func (store *Store) LoadPlainFile(in []byte) (sops.TreeBranch, error) {
-	var data yaml.MapSlice
-	if err := (yaml.CommentUnmarshaler{}).Unmarshal(in, &data); err != nil {
+func (store *Store) LoadPlainFile(in []byte) (sops.TreeBranches, error) {
+	var data []yaml.MapSlice
+	if err := (yaml.CommentUnmarshaler{}).UnmarshalDocuments(in, &data); err != nil {
 		return nil, fmt.Errorf("Error unmarshaling input YAML: %s", err)
 	}
-	return store.mapSliceToTreeBranch(data), nil
+
+	var branches sops.TreeBranches
+	for _, doc := range data {
+		branches = append(branches, store.mapSliceToTreeBranch(doc))
+	}
+	return branches, nil
 }
 
 func (store *Store) EmitEncryptedFile(in sops.Tree) ([]byte, error) {
-	yamlMap := store.treeBranchToYamlMap(in.Branch)
-	yamlMap = append(yamlMap, yaml.MapItem{Key: "sops", Value: stores.MetadataFromInternal(in.Metadata)})
-	out, err := (&yaml.YAMLMarshaler{Indent: 4}).Marshal(yamlMap)
-	if err != nil {
-		return nil, fmt.Errorf("Error marshaling to yaml: %s", err)
+	out := []byte{}
+	var err error
+	if len(in.Branches) > 1 {
+		for i, branch := range in.Branches {
+			if i > 0 {
+				out = append(out, "---\n"...)
+			}
+			yamlMap := store.treeBranchToYamlMap(branch)
+			yamlMap = append(yamlMap, yaml.MapItem{Key: "sops", Value: stores.MetadataFromInternal(in.Metadata)})
+			tout, err := (&yaml.YAMLMarshaler{Indent: 4}).Marshal(yamlMap)
+			if err != nil {
+				return nil, fmt.Errorf("Error marshaling to yaml: %s", err)
+			}
+			out = append(out, tout...)
+		}
+	} else {
+		yamlMap := store.treeBranchToYamlMap(in.Branches[0])
+		yamlMap = append(yamlMap, yaml.MapItem{Key: "sops", Value: stores.MetadataFromInternal(in.Metadata)})
+		out, err = (&yaml.YAMLMarshaler{Indent: 4}).Marshal(yamlMap)
+		if err != nil {
+			return nil, fmt.Errorf("Error marshaling to yaml: %s", err)
+		}
 	}
 	return out, nil
 }
 
-func (store *Store) EmitPlainFile(in sops.TreeBranch) ([]byte, error) {
-	yamlMap := store.treeBranchToYamlMap(in)
-	out, err := (&yaml.YAMLMarshaler{Indent: 4}).Marshal(yamlMap)
-	if err != nil {
-		return nil, fmt.Errorf("Error marshaling to yaml: %s", err)
+func (store *Store) EmitPlainFile(tree sops.Tree) ([]byte, error) {
+	var out []byte
+	if len(tree.Branches) > 1 {
+		for i, branch := range tree.Branches {
+			if i > 0 {
+				out = append(out, "---\n"...)
+			}
+			yamlMap := store.treeBranchToYamlMap(branch)
+			tmpout, err := (&yaml.YAMLMarshaler{Indent: 4}).Marshal(yamlMap)
+			if err != nil {
+				return nil, fmt.Errorf("Error marshaling to yaml: %s", err)
+			}
+			out = append(out[:], tmpout[:]...)
+		}
+	} else {
```

Comment:
Same as above, I'm not sure why we need to handle the 'single-branch' case separately.